pkg/appsec/api/http/mux.go

```diff
@@ -0,0 +1,18 @@
+package http
+
+import (
+	"net/http"
+
+	"github.com/DataDog/datadog-agent/pkg/appsec/api/http/v0_1_0"
+	"github.com/DataDog/datadog-agent/pkg/appsec/config"
+)
+
+// NewServeMux creates and returns the HTTP request multiplexer serving the
+// AppSec HTTP API. The API is versioned according to the X-Api-Version header
+// value so that requests are routed to the request multiplexer of the
+// given version.
+func NewServeMux(cfg *config.Config) *http.ServeMux {
```

Comment:
This is overkill. You have a proxy inside a mux, inside another mux. There are no routes, there is just one proxy. Can we just drop everything and leave only the proxy? In `pkg/trace/endpoints.go` you can then link it straight to `/appsec/proxy`.

Reply:
Done: now simplified to a single`appsec` package.